### DISCUSSION TITLE
[ FF7 ] Strange transparency behaviour using Ninostyle Chibi models and SYW Upscaled Backgrounds

### DISCUSSION THREAD

**Discussion by LaZar00** [julianxhokaxhiu/FFNx#464] [Ideas]
Description:

I have detected when using Ninostyle Chibi models that sometimes you can see the background in the transparent part of the eyes of the model, if the eyes are a .tex file.

As a sample to check, you can use the model BPDC.HRC with its .tex files, the upscaled textures for the field onna_52 of SYW 5.22 and go to the Honey Bee Inn, the bottom left door.

I attach the model file, the field file and the save01 of the game that has the savegame selected (as "Mercado Muro" (spanish) or "Wall Market") with Level 10.

You must put the model in _direct\char_ folder, the field in _mod\Textures_ folder and the save in the save folder. Then run the game as usual.

As a screenshot, this is the result where you can see the bug (look at the eyes of the guy with red gym suit), where I used the Ninostyle Chibi and upscaled SYW texture for the field onna_52. You can see a "blue" border around the eye, which I assume is the background:
![imagen](https://user-images.githubusercontent.com/46090031/177385079-171636fe-f59d-4f63-8d98-f9762387a13d.png)

And this screenshot is only the Ninostyle Chibi model with vanilla field background:
![imagen](https://user-images.githubusercontent.com/46090031/177385195-8c4ab59a-0c25-4e6d-96ca-051809d9f69d.png)

For the test I used DirectX12, Fullscreen, 1920x1080, AR 4:3, VSync disabled, NO MSAA and AF/Lighting does not matters

[BPDC.HRC.zip](https://github.com/julianxhokaxhiu/FFNx/files/9048381/BPDC.HRC.zip)
[field_onna_52.zip](https://github.com/julianxhokaxhiu/FFNx/files/9048383/field_onna_52.zip)
[save01.zip](https://github.com/julianxhokaxhiu/FFNx/files/9048385/save01.zip)

**Comment by julianxhokaxhiu** [comment]
Allright here are some findings I've made around this, which is not a FFNx bug but more probably how the model is made.

You can see that the draw call order is wrong logically, before is drawing the eyes and then the face. When eyes are drawn are blending with what is behind at that moment, which is the blue background. When the fact is drawn it is too late so eyes are going to have the blue border.

Not sure how the order is being inherited but I assume is based on how the model is made.

Eye 1:
![image](https://user-images.githubusercontent.com/1237070/177833012-a412ed93-2d47-4a4f-bf74-316b5bf500f3.png)

Eye 2:
![image](https://user-images.githubusercontent.com/1237070/177833037-7df59119-c62d-4315-8fad-c771af5e33e7.png)

Face:
![image](https://user-images.githubusercontent.com/1237070/177833058-c4eb6f46-2096-4cc2-9191-0e86899cf769.png)

RenderDoc scene: 
[scene.zip](https://github.com/julianxhokaxhiu/FFNx/files/9066058/scene.zip) ( rename to .7z and extract with 7-Zip )

**Comment by LaZar00** [comment]
It is ok for me. I understand the different blending FFNx does when using external textures and maybe it is a consequence of provoking this eye thing because we have to change the order of the polys that have to be rendered of the poly, in the head part, because (again) they use textures in the face (when in vanilla the models only use vertex/poly colors).

I would prefer not to close this yet. But I will leave it up to you.

**Comment by LaZar00** [comment]
Ok, I have tested some more things. For example, I have added a square (2 triangles) behind the eye texture groups (the red one):
![imagen](https://user-images.githubusercontent.com/46090031/177957850-766c4c37-1581-4710-8b84-f46d06e425bf.png)

This has not texture, is vertex/poly colored.
If we assume the order of rendering is the groups order (as RenderDoc has demostrated), the first rendering group would be this red square. So, we would see in red the blue glow thing around the eye. But as this is rendered, not texture blended, the result in game is the same as always. The eye has the blue glow of the background in the transparency.

Is this normal?
[BPDC.zip](https://github.com/julianxhokaxhiu/FFNx/files/9070631/BPDC.zip)

I attach the model if anyone wants to make some tests.

PD: I checked RenderDoc (yep, finally I have installed it) and it seems NOT to render the red square, or is transparent. WHY? The next two events are both eyes which are rendered as usual and described by julianxhokaxhiu.
![imagen](https://user-images.githubusercontent.com/46090031/177965808-a1f8b0fe-e821-466c-8d0e-e5dc6a81b719.png)

**Comment by LaZar00** [comment]
Ok, more testing about this issue. Finally I have made possible to show a back triangle (only vertex colored):
![imagen](https://user-images.githubusercontent.com/46090031/178014192-ad9e4700-d43a-49e5-aef5-475c549db243.png)

Then we have the eyes:
![imagen](https://user-images.githubusercontent.com/46090031/178014227-c608b3dd-d6ed-4def-9aba-a3529ba6c54d.png)

![imagen](https://user-images.githubusercontent.com/46090031/178014249-42f9936c-577e-4628-b58d-9c5ba80fec47.png)

And finally the head:
![imagen](https://user-images.githubusercontent.com/46090031/178014374-ea21a506-9c55-46bf-b69c-2b8aeaa990f3.png)


All the polys are in the same .P Model. This is the configuration of the .P Model:
![imagen](https://user-images.githubusercontent.com/46090031/178016013-a3cda437-413b-45ac-a656-2162f981ec34.png)


Here we have:
Group 00: NO TEXTURED. Vertex/Poly colored. I put red to see the poly in RenderDoc, but we could put it as black (or maybe better near black, like R16,G10,B10).
Group 01: TEXTURE: Left eye.
Group 02: TEXTURE: Right eye.
Group 03: TEXTURE: Rest of the head (hair, face, etc...)

This solution seems to avoid to put the color of the background in the eye transparency AND blink the eyes.
We need to use the Group 00 properties this way (look the "True" checkbox of TEXTURE checked while the own TEXTURE checkbox is unchecked):
![imagen](https://user-images.githubusercontent.com/46090031/178015409-d8f8743c-70ac-4879-9150-61c35a6d10fe.png)
